src/pybind/kaldi_pybind.cc

```diff
@@ -45,8 +43,21 @@ void sequential_matrix_reader(py::module &m, std::string suffix) {
 }
 
 template<class Holder>
-void matrix_writer(py::module &m, std::string suffix) {
-  std::string name = "BaseFloatMatrixWriter" + suffix;
+void random_access_matrix_reader(py::module &m, std::string data_type) {
+    std::string name = "RandomAccessBaseFloat" + data_type + "Reader";
+    py::class_<RandomAccessTableReader<Holder > >(m, name.c_str())
+        .def(py::init<>()) 
+        .def(py::init<const std::string &>()) 
+        .def("Open", &RandomAccessTableReader<Holder>::Open) 
```

Comment:
Should some macros be defined to help such repetitive tasks?

For instance

```cpp
#define DEF_CLASS(name) py::class_<PyClass> cls(m, name.c_str())
#define DEF(name, method) cls.def(name.c_str(), &PyClass::method)
```
where `PyClass` is defined in a specific function.

For example, for `random_access_matrix_reader`,

```cpp
{
using PyClass = RandomAccessTableReader<Holder>;
std::string name = "RandomAccessBaseFloat" + data_type + "Reader";
DEF_CLASS(name);
DEF(name, Open);
DEF(name, IsOpen);
....
}
```

For `matrix_writer`,
```cpp
{
using PyClass = TableWriter <Holder>;
std::string name = "BaseFloat" + data_type + "Writer";
DEF_CLASS(name.c_str())
DEF(name, Open);
DEF(name, IsOpen);
....
}
```

For the constructors, similar macros can be defined.
If a function requires reference semantics, more macros can be defined.